cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -373,6 +377,24 @@ func gcloudProject() []string {
 	return []string{data.Core.Project}
 }
 
+func setupTermDelay() {
+	terminated := make(chan bool, 0)
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+
+	go func() {
+		<-terminated
+		time.Sleep(*termDelay)
```

Comment:
This is rather pessimistic, it'd be nice if the client proxy shut off as soon as there were no connections...

Instead of just always sleeping for termDelay, what do you think of adding a ticker that checks if client.ConnectionsCounter == 0, and if so, calls os.Exit(0) early? (If you go that route, take care to also remove the `if c.MaxConnections > 0` that only updates the ConnectionsCounter if needed.